File: Neato-SDK/neato-sdk-android/src/main/java/com/neatorobotics/sdk/android/authentication/AccessTokenDatasource.java
package com.neatorobotics.sdk.android.authentication;

import java.util.Date;

/**
 * Neato-SDK
 * Created by Marco on 06/05/16.
 * Copyright © 2016 Neato Robotics. All rights reserved.
 */
public interface AccessTokenDatasource {
    void storeToken(String token, Date expires);
    String loadToken();
    void clearToken();
    boolean isTokenValid();
}

File: Neato-SDK/neato-sdk-android/src/main/java/com/neatorobotics/sdk/android/authentication/AccessTokenDatasource.kt
/*
 * Copyright (c) 2019.
 * Neato Robotics Inc.
 */

package com.neatorobotics.sdk.android.authentication

import java.util.Date

interface AccessTokenDatasource {
    val isTokenValid: Boolean
    fun storeToken(token: String, expires: Date)
    fun loadToken(): String?
    fun clearToken()
}
